key/key.go

```diff
@@ -0,0 +1,13 @@
+package key
+
+import (
+	hapichart "k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+func ChartVersion(chart *hapichart.Chart) string {
+	if chart.Metadata == nil {
+		return ""
+	}
+
+	return chart.Metadata.Version
+}
```

Comment:
Just adding ChartVersion for now. We can add more helpers as needed.

Reply:
I would prefer to return custom tailored type instead of having extractors here. Something like https://github.com/giantswarm/devctl/blob/be38aacbff4b0409211ad2fd36e5803b0f99cc58/pkg/githubclient/types.go#L11.

We could have a `Chart` type with `Version` field. The usage would be obvious then. We can add fields as we need then and don't need to import anything outside library.

Reply:
Yes that's how I had it originally. The problem is the `Metadata` struct is large and we will need a lot of values from it. This is because we want the `Chart.yaml` to be the master copy for metadata about charts.

https://github.com/helm/helm/blob/4634f5f2faa0caa7de390c95c7d27be22d56a156/pkg/proto/hapi/chart/metadata.pb.go#L75-L112

Reply:
I'd like to get more feedback on this. @kopiczko, your point on not needing to import any helm packages is a good one. My concern is just that we'll need to end up replicating the protobufs struct.

Reply:
When I was developing this github package I also did a lot of nil checking (and possibly other validation) which is madness with protobuf generated code so we don't have to repeat everywhere. Something you have there `chart == nil || chart.Metadata == nil`.

Will we ever use more than 10% of those fields? And is it really an overhead with maintaining a struct when we have an extractor for everything anyway? I think it is actually easier. It's super easy to get lost in all that extractors. We every so often notice that we don't use some for a long time in our operators.

Reply:
We will definitely use more than 10% of the metadata fields. But I agree the protobuf structs are hard to work with. Cool I'll define a struct in spec.go and go with that approach.

Reply:
I'd prefer (kind of obvious - this is how I did it) to have that in `types.go`. I don't see why would we pollute `spec.go` with private constructors and validations. `spec.go` declare returned types and I think it's enough to follow what's going on. Especially given that we often return types from outside the package.